radix-engine-common/src/data/scrypto/custom_well_known_types.rs

```diff
@@ -228,38 +311,364 @@ create_well_known_lookup!(
+                    (
+                        2u8,
+                        named_tuple("CountOf", [U8_TYPE, RESOURCE_OR_NON_FUNGIBLE_LIST_TYPE])
+                    ),
+                    (
+                        3u8,
+                        named_tuple("AllOf", [RESOURCE_OR_NON_FUNGIBLE_LIST_TYPE])
+                    ),
+                    (
+                        4u8,
+                        named_tuple("AnyOf", [RESOURCE_OR_NON_FUNGIBLE_LIST_TYPE])
+                    ),
+                ],
+            )
+        ),
+        (
+            RESOURCE_OR_NON_FUNGIBLE,
+            ROLE_ASSIGNMENT_TYPES_START + 4,
+            named_enum(
+                "ResourceOrNonFungible",
+                [
+                    (
+                        0u8,
+                        named_tuple("NonFungible", [NON_FUNGIBLE_GLOBAL_ID_TYPE])
+                    ),
+                    (1u8, named_tuple("Resource", [RESOURCE_ADDRESS_TYPE])),
+                ],
+            )
+        ),
+        (
+            RESOURCE_OR_NON_FUNGIBLE_LIST,
+            ROLE_ASSIGNMENT_TYPES_START + 5,
+            array_of(RESOURCE_OR_NON_FUNGIBLE_TYPE)
+        ),
+        (
+            OWNER_ROLE,
+            ROLE_ASSIGNMENT_TYPES_START + 6,
+            named_enum(
+                "OwnerRole",
+                [
+                    (0u8, named_tuple("None", [])),
+                    (1u8, named_tuple("Fixed", [ACCESS_RULE_TYPE])),
+                    (2u8, named_tuple("Updatable", [ACCESS_RULE_TYPE])),
+                ],
+            )
+        ),
+        (
+            ROLE_KEY,
+            ROLE_ASSIGNMENT_TYPES_START + 7,
+            named_transparent("RoleKey", string_type_data(),)
+        ),
+        // OTHER MODULE TYPES
+        (
+            OBJECT_MODULE_ID,
+            OTHER_MODULE_TYPES_START + 0,
+            named_enum(
+                // This is a clearer name for UIs, but not time to change in the code right now
+                "ModuleId",
+                [
+                    (0u8, named_tuple("Main", [])),
+                    (1u8, named_tuple("Metadata", [])),
+                    (2u8, named_tuple("Royalty", [])),
+                    (3u8, named_tuple("RoleAssignment", [])),
+                ],
+            )
+        ),
+        (
+            MODULE_ID,
+            OTHER_MODULE_TYPES_START + 1,
+            named_enum(
+                // This is a clearer name for UIs, but not time to change in the code right now
+                "NativeModuleId",
+                [
+                    (1u8, named_tuple("Metadata", [])),
+                    (2u8, named_tuple("Royalty", [])),
+                    (3u8, named_tuple("RoleAssignment", [])),
+                ],
+            )
+        ),
+        (
+            ROYALTY_AMOUNT,
+            OTHER_MODULE_TYPES_START + 2,
+            named_enum(
+                "RoyaltyAmount",
+                [
+                    (1u8, named_tuple("Free", [])),
```

Comment:
Good catch!!

Hah 🤦 my added tests failed to catch this because XRD and USD variants have the same signature. Will update the tests, and the enum.

Reply:
Thankfully other tests failed because of it - although only the `no_std` and `wasmer` tests, not the main `release` tests (!??) https://github.com/radixdlt/radixdlt-scrypto/actions/runs/6069142553/job/16463099325?pr=1439